resolver/src/resolver_future.rs

```diff
@@ -188,6 +188,12 @@ impl ResolverFuture {
             let mut names =
                 Vec::<Name>::with_capacity(1 /*FQDN*/ + 1 /*DOMAIN*/ + self.config.search().len());
 
+            // if not meeting ndots, we always do the raw name in the final lookup
+            // note: number of dots will always be one less than the number of labels
+            if name.num_labels() as usize <= self.options.ndots && !name.is_localhost() {
```

Comment:
I suggest that you factor out `let raw_name_first = name.num_labels() as usize <= self.options.ndots && !name.is_localhost();` into a variable, and then test `raw_name_first` here and then `!raw_name_first` below, to make it clear how these two are related. Then you can also remove the duplication in the comments.

Reply:
Yes. That felt messy. Not sure why I didn’t do that initially.